src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -63,6 +67,20 @@ private static String encodeValue(String value) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
+    String redirectPath = request.getRequestURI();
+    if (request.getQueryString() != null) {
+      redirectPath += "?" + request.getQueryString();
+    }
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      request.setAttribute(AUTH_URL, userService.createLogoutURL(redirectPath));
+      request.setAttribute(AUTH_LABEL, "Logout");
+    } else {
+      request.setAttribute(AUTH_URL, userService.createLoginURL(redirectPath));
+      request.setAttribute(AUTH_LABEL, "Login");
```

Comment:
change login/logout to constant

Reply:
I'm actually okay with it here because it's not like the other cases - this is the only place it should ever be used in this file.

(In the other cases where I've asked strings to be made private constants, it was because it was being used as a constant identifier in multiple places, such as as a query parameter or a database key, etc.)